mantidproject/mantid
# Project Recovery: Handle live data streams

**Description of work.**
Project Recovery will now successfully restart lost live data streams on recovery

**To test:**
- Start Mantid
- Start LiveData
- Wait for LiveData to have created the workspace
- Run `mantidplot.app.saveRecoveryCheckpoint()` or wait for project recovery to save a checkpoint
- Run Segfault
- Start mantid and click yes when asked to recover
- Notice how it will then start up the stream again



Fixes #22825  


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
